### ISSUE TITLE
Tensorforce network loss...

### ISSUE THREAD

**Issue by ZeroMaxinumXZ** [tensorforce/tensorforce#582]
Hi. I'm looking for a way to get the loss of a Tensorforce network. Is this possible? Thanks.

**Comment by AlexKuhnle** [comment]
Hey! If you want to display the loss, the easiest way is TensorBoard and summaries. For that, I would ideally recommend switching to the `major-revision` branch, which is a new version handling summaries much better, but just simple loss should also work with the `master` branch. Regarding getting the model, do you mean you want to store the weights? Again, the easiest way is to use the TensorFlow functionality, supported via `save_model` (or I think only `save` on the new branch).

**Comment by AlexKuhnle** [comment]
Let me know if you need more information on that.

**Comment by ZeroMaxinumXZ** [comment]
I just want the loss as a list... Basically, what I want to do is some kind of an AutoML approach, where the loss of a tensorforce model is fed as a reward to another agent which builds the network for the first agent. Also is there any sort of documentation for the major-revision branch of Tensorforce I can look at?

**Comment by ZeroMaxinumXZ** [comment]
@AlexKuhnle Nevermind, I see that you already kind-of have an AutoML thing in the major-revision... Sorry.

**Comment by ZeroMaxinumXZ** [comment]
@AlexKuhnle, oh, you have AutoML in the revision... However I do want to try my own approach and see how it compares...

**Comment by AlexKuhnle** [comment]
Right now there is no particular documentation of the changes (but the act/observe API also hasn't really changed). However, there will be another update (with more internal changes) soon, which includes a bit more documentation. Where do you feel you need documentation in particular?

The new branch still accepts the same network configurations as before, just additionally has an automatically configured network (not quite the same as AutoML, I think ;-).

In the new branch, there is also ways to query the values of tensors (`query=` argument), so that may help to achieve what you want. Check [this](https://github.com/tensorforce/tensorforce/blob/major-revision/tensorforce/tests/agent_unittest.py#L110). I can write more if that is something of interest, because right now it does require going through code... :-/

**Comment by ZeroMaxinumXZ** [comment]
Nevermind on the documentation, it's perfect, don't know where I was going with that...:)

Please tell me more about the `query=` argument... Don't understand how to get the loss from that... Thank you. :)

**Comment by AlexKuhnle** [comment]
I hope the test gives an idea how it's supposed to work, but the actual retrievable tensors have to be increased. I should be able to add that soon. In fact, this functionality should be made more easily accessible in general.

**Comment by ZeroMaxinumXZ** [comment]
@AlexKuhnle I agree that this functionality should be made more accessible. Maybe a model.get_loss() method... Like in Keras?

**Comment by AlexKuhnle** [comment]
It will have to go via the `query` argument as it happens currently, but I'm thinking of something like `agent.get_retrievable_tensors()` or so, to programmatically explore what tensors can be queried, or so. Maybe this is what you mean...

It will take a few days, I fear, as this will have to happen as part of the finalized revision version, so I need to get that pushed first.
